### ISSUE TITLE
Сделать отдельную библиотеку универсальных тасков для opm

### ISSUE THREAD

**Issue by artbear** [oscript-library/opm#47]
Нужен набор универсальных тасков для opm
+ запуск  1testrunner для тестирования проекта
+ запуск 1bdd для прогона всех фич проекта
+ запуск 1bdd для прогона одной фичи
+ запуск 1bdd для генерации шагов фичи
+ создание пакета библиотеки для публикации на хабе
+ проверка генерации пакета библиотеки 
+ проверка установки пакета
+ новые шаги ...

Сейчас приходится дублировать набор этих задач в каждом проекте. Причем для разных зада нужен индивидуальный набор задач.

Чтобы не нагружать движок opm, предлагаю выделить отдельную библиотеку opm-tasks

**Comment by artbear** [comment]
Создал заготовку https://github.com/oscript-library/opm-tasks

**Comment by nixel2007** [comment]
Помимо этого - в движке opm предусмотреть возможность расширения списка тасков с внешнего расположения.
Возможно разработать некий механизм пакетов-плагинов, который будет добавлять новые таски в список доступных для запуска.

**Comment by artbear** [comment]
Также мне нравится идея с регистрацией внешних тасков

Или https://github.com/oscript-library/gitsync/pull/45#issuecomment-280754954
>+ `opm register task create template-creator.os`
>+ `opm run create lib`
>
>в рамках регистрации произвольных задач для opm

**Comment by EvilBeaver** [comment]
лучше своровать из yeoman

    opm new application-template

**Comment by artbear** [comment]
@EvilBeaver `opm new application-template`
это команда на создание таска или все-таки на создание шаблона библиотеки/приложения, что является назначением команды `opm prepare` ?

**Comment by EvilBeaver** [comment]
Да, это развитие opm-prepare. т.е. мы генерим новый проект по шаблону приложения.
prepare по сути - это абстрактный пакет. а Шаблон - это уже определенный фреймворк с командами.

**Comment by artbear** [comment]
первое предложение частично понял, частично нет.
2-е предложение вообще не понял :)
Опиши на коротких примерах, плиз, что должна выполнять команда `opm prepare`, а что команда `opm new` (если эта команда будет).
Жду

**Comment by EvilBeaver** [comment]
opm - это встроенная команда, создающая заготовку пакета.
opm new - это внешний таск. Такой же, как test и др. (см. тему данной issue)

**Comment by asosnoviy** [comment]
Апну ибо накипело.  
На текущий момент есть:
2 стабильных таска: `test.os` `coverage.os`  
Грубым напильником `ЗапускательЗадач` обучается искать таски не только в `\tasks`, но и в относительном каталоге opm.  

Итого: профит есть, красоты нет...
Как получить путь к каталогу библиотеки?

**Comment by asosnoviy** [comment]
@artbear мой последний, это первый попавшийся из репозиториев @nixel2007 =))  
https://github.com/oscript-library/irac/blob/master/tasks/test.os  
Вот какой-то.
С меня WIP. Дальше на коде обсудим

**Comment by artbear** [comment]
@asosnoviy Это очень старый вариант :(

В нем есть недостатки:
+ нет поддержки фич 1bdd одновременно с наличием тестов 1testrunner
+ жесткая привязка к каталогу
+ не возвращается код возврата
+ и что-то еще

я в packman вроде бы добавлял последний актуальный
